sdk/core/core-util/src/checkEnvironment.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+declare global {
+  interface Window {
+    document: unknown;
+  }
+
+  interface DedicatedWorkerGlobalScope {
+    constructor: {
+      name: string;
+    };
+  }
+
+  interface Navigator {
+    product: string;
+  }
+
+  interface DenoGlobal {
+    version: {
+      deno: string;
+    };
+  }
+
+  interface BunGlobal {
+    version: string;
+  }
+
+  const window: Window;
+  const self: DedicatedWorkerGlobalScope;
+  const Deno: DenoGlobal;
+  const Bun: BunGlobal;
+  const navigator: Navigator;
+}
+
+/**
+ * A constant that indicates whether the environment the code is running is a Web Browser.
+ */
+export const isBrowser = typeof window !== "undefined" && typeof window.document !== "undefined";
+
+/**
+ * A constant that indicates whether the environment the code is running is a Web Worker.
+ */
+export const isWebWorker =
+  typeof self === "object" && self.constructor?.name === "DedicatedWorkerGlobalScope";
+
+/**
+ * A constant that indicates whether the environment the code is running is Node.JS.
+ */
+export const isNode =
```

Comment:
Also, I think you may have to have counterpart `checkEnvironment.browser.ts` with `export const isNode = false` to not break the browser?

similarly for deno and bun?

Reply:
There is no need to have a separate file for the browser, as all of these checks should be valid for any environment.